### PR TITLE
Add autocompletion for defined functions and constants

### PR DESCRIPTION
**Pull request by gchaincl**
Just working for PHP built in functions

### CODE REVIEW

File: lib/Boris/Autocompletion.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        return $this->symbols;
+    }
+
+    private function getVariables() {
+        /**
+         * TODO: Find a way to access EvalWorker context/scope to access vars
+         */
+        return ['this', 'container'];
```
Comment (tejasmanohar) [inline lib/Boris/Autocompletion.php]: We do want to maintain compatibility with PHP 5.3+ for a few more years, at least. @gchaincl if you fix this, we will merge. :tada:

**Comment by tejasmanohar** [comment]
This is awesome, @gchaincl! Thanks for updating to old array syntax for older versions of PHP :+1:  - Merged.

File: lib/Boris/Autocompletion.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+/* vim: set shiftwidth=2 expandtab softtabstop=2: */
+
+namespace Boris;
+
+class Autocompletion {
+    private $symbols = [];
```
Comment (marcosdsanchez) [inline lib/Boris/Autocompletion.php]: Old array syntax should be used here too.

**Comment by tejasmanohar** [comment]
@marcosdsanchez resolved - fabd509f0a96669e98f38bc7afcf7b1465e183f1

**Comment by yuriteixeira** [comment]
Wow! We really got this? Trying NOW!

**Comment by tejasmanohar** [comment]
This was removed from the core of Boris. It's too heavy for a light-weight "Tiny REPLY for PHP" (our tagline), and @d11wtq's vision for the project. We are investigating a plugin API that would make Boris extendable for things like this :+1:

**Comment by gchaincl** [comment]
@tejasmanohar what do you mean with "to heavy", any performance test for that?
A plugin API will be nice :+1:

**Comment by tejasmanohar** [comment]
@gchaincl just that the original intention of boris was to keep it super light-weight and tiny, in features, code size, etc. just a basic REPL. but we're looking to make a plugin API for boris to allow extensions that aren't quite one-size-fit-all basic REPL needs :)

**Comment by gchaincl** [comment]
fair enough. thanks

**Comment by tejasmanohar** [comment]
@gchaincl are you collaborating on the plugin API? boris is looking for more contributors, btw

**Comment by gchaincl** [comment]
thanks, but I don't have enough time.

**Pull request merged**